src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
+  private logger = new Logger();
+
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<Response<T>> {
+    const statusCode = context.switchToHttp().getResponse().statusCode;
+    const req = context.switchToHttp().getRequest();
+    const userDetail = req?.user || {};
+
+    return next.handle().pipe(
+      map((data) => {
+        const response = {
+          isSuccess: statusCode >= 200 && statusCode < 300 ? true : false,
+          statusCode: statusCode,
+          error: data && data.message ? data.message : '',
+          message: data && data.message ? data.message : '',
+          data: data && data.message ? { ...data, message: undefined } : data,
+        };
+        this.logger.debug(
+          `Response: [${JSON.stringify(
+            response,
+          )}] || User Details: [${JSON.stringify(userDetail)}]`,
+        );
+
+        return response;
+      }),
+    );
```

Comment:
PullProofAI

The logic in the `map` function can be simplified by removing redundant checks and improving readability. Consider refactoring as follows:

```typescript
return next.handle().pipe(
  map((data) => {
    const isSuccess = statusCode >= 200 && statusCode < 300;
    const response = {
      isSuccess,
      statusCode,
      error: data?.message || '',
      message: data?.message || '',
      data: data ? { ...data, message: undefined } : data,
    };
    this.logger.debug(`Response: [${JSON.stringify(response)}] || User Details: [${JSON.stringify(userDetail)}]`);
    
    return response;
  }),
);
```

### Summary
The `ResponseInterceptor` class could benefit from some comments for clarity. Additionally, the logic within the `map` function can be simplified for better readability and maintenance.


<!-- This is an auto-generated comment by PullProofAI -->